pkg/syncer/metadatasyncer.go

```diff
@@ -253,17 +261,71 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 	fullSyncTicker := time.NewTicker(time.Duration(getFullSyncIntervalInMin(ctx)) * time.Minute)
 	defer fullSyncTicker.Stop()
 	// Trigger full sync
-	go func() {
-		for ; true; <-fullSyncTicker.C {
-			ctx, log = logger.GetNewContextWithLogger()
-			log.Infof("fullSync is triggered")
-			if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorGuest {
-				pvcsiFullSync(ctx, metadataSyncer)
-			} else {
-				csiFullSync(ctx, metadataSyncer)
-			}
+	// If TriggerCsiFullSync feature gate is enabled, use TriggerCsiFullSync to trigger
+	// full sync. If not, directly invoke full sync methods.
+	if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.TriggerCsiFullSync) {
+		log.Infof("%q feature flag is enabled. Using TriggerCsiFullSync API to trigger full sync",
+			common.TriggerCsiFullSync)
+		// Get a config to talk to the apiserver
+		restConfig, err := config.GetConfig()
+		if err != nil {
+			log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+			return err
 		}
-	}()
+
+		cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+		if err != nil {
+			log.Errorf("Failed to create CnsOperator client. Err: %+v", err)
+			return err
+		}
+		go func() {
+			for ; true; <-fullSyncTicker.C {
+				ctx, log = logger.GetNewContextWithLogger()
+				log.Infof("periodic fullSync is triggered")
+				triggerCsiFullSyncInstance, err := getTriggerCsiFullSyncInstance(ctx, cnsOperatorClient)
+				if err != nil {
+					log.Warnf("Unable to get the trigger full sync instance. Err: %+v", err)
+					continue
+				}
+
+				// Update TriggerCsiFullSync instance if full sync is not already in progress
+				if triggerCsiFullSyncInstance.Status.InProgress {
+					log.Infof("There is a full sync is already in progress. Ignoring this current cycle of periodic full sync")
+				} else {
+					triggerCsiFullSyncInstance.Spec.TriggerSyncID = triggerCsiFullSyncInstance.Spec.TriggerSyncID + 1
+					err = updateTriggerCsiFullSyncInstance(ctx, cnsOperatorClient, triggerCsiFullSyncInstance)
+					if err != nil {
+						log.Errorf("Failed to update TriggerCsiFullSync instance: %q to increment the TriggerFullSyncId. Error: %v",
```

Comment:
I think it is good to log the entire `triggerCsiFullSyncInstance` object and will be useful for debugging